examples/vision/video_classification.py

```diff
@@ -0,0 +1,469 @@
+
+    outputs = feature_extractor(preprocessed)
+    return keras.Model(inputs, outputs, name="feature_extractor")
+
+
+"""
+We now create TensorFlow dataset objects from the video frame paths to efficiently
+extract their features.
+"""
+
+
+def load_image(image_path):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, 3)
+    image = tf.image.resize(image, (IMG_SIZE, IMG_SIZE))
+    return image
+
+
+train_images_list = new_train_df["image_path"].values.tolist()
+train_ds = tf.data.Dataset.from_tensor_slices(train_images_list)
+train_ds = train_ds.map(
+    load_image, num_parallel_calls=tf.data.AUTOTUNE, deterministic=True
+).batch(BATCH_SIZE)
+
+test_images_list = new_test_df["image_path"].values.tolist()
+test_ds = tf.data.Dataset.from_tensor_slices(test_images_list)
+test_ds = test_ds.map(
+    load_image, num_parallel_calls=tf.data.AUTOTUNE, deterministic=True
+).batch(BATCH_SIZE)
+
+feature_extractor = build_feature_extractor()
+train_features = feature_extractor.predict(train_ds)
+test_features = feature_extractor.predict(test_ds)
+
+"""
+## Reshape features and prepare labels
+
+Now that we have extracted the frame features, we need to feed them to our recurrent
+network considering the temporality present inside of them. Data fed to the recurrent
+layers must have a time-step dimension. So, how do we incorporate that in the extracted
+features?
+
+The features have a shape of `len(train_frames), 5, 5, 2048`). Recall that each video is
+now represented as an ordered sequence of **five** frames. So, we can reshape our features
+like so: `len(train_frames)/5, 5, 5*5*2048`). The second dimension (5) is now the
+time-step dimension. It turns out that this heuristic works quite well in practice. It
+just needs as adjustment in the number of examples such that it is divisible by the
+number of frames representing a video. Fortunately, for this example, we did not have to
+do that explicitly.
+"""
+
+train_features = train_features.reshape(
+    len(train_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+test_features = test_features.reshape(
+    len(test_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+
+"""
+We now need to prepare the labels i.e. the action classes which are present as strings.
+Neural networks do not understand string values, they must be converted to some numerical
+form before they are fed to neural networks. Here we will use the [`StringLookup`](https://keras.io/api/layers/preprocessing_layers/categorical/string_lookup)
+layer to binarize the string class labels. It will first give us an integer encoding of
+the string values and then it will [one-hot encode](https://developers.google.com/machine-learning/glossary#one-hot-encoding)
+those values. The point of making the values one-hot encoded will be made clear in a moment. 
+"""
+
+label_processor = keras.layers.experimental.preprocessing.StringLookup(
+    max_tokens=6, num_oov_indices=0, output_mode="binary", sparse=False
```

Comment:
Referring to [this guide](https://www.tensorflow.org/tutorials/structured_data/preprocessing_layers#categorical_columns) it seems like we need to first run the labels `StringLookup` and then use `CategoryEncoding` since `CategoryEncoding` cannot directly operate strings. This is why I directly used `StringLookup`.

Reply:
I was confused, you're right. It looks good.

Since you're calling `adapt` to find out the label set, then you don't need to specify `max_tokens` (also `sparse=False` is the default so no need to specify it).

Another thing you could do is skip calling `adapt` altogether, by doing:

```python
label_processor = keras.layers.experimental.preprocessing.StringLookup(
    num_oov_indices=0, output_mode="binary",
    vocabulary=new_train_df["class"].values
)
```

That way the layer will already have its vocab ready and won't need `adapt` anymore.

Reply:
Great. Will apply this right away.

Reply:
@fchollet, after trying the suggestion, I got this:

```
ValueError: The passed vocabulary has at least one repeated term. Please uniquify your dataset. The repeated terms are ['CricketShot', 'PlayingCello', 'Punch', 'ShavingBeard', 'TennisSwing']
```